### DISCUSSION TITLE
Preview of legacy archve import (v36.1 beta)

### DISCUSSION THREAD

**Discussion by chylex** [chylex/Discord-History-Tracker#185] [General]
This is a beta release that includes the possibility to import legacy archives created by browser-only versions of DHT.

When you import a legacy archive, you will see a dialog that asks you to fill in the IDs of servers, groups, and direct messages. Filling in the IDs is optional (and not possible if a server no longer exists). If you don't fill in an ID, the app will generate a random one, but I cannot guarantee that you won't run into issues if you continue tracking servers that you imported with random IDs.

The legacy archives are also missing attachment IDs and sizes, but asking you to fill those in would be ridiculous. This means that for ex. in the new Attachments tab, you will see that the total size says `0 B`. If you download the attachments, the Attachments tab will report the actual size then. You can also re-track imported messages if possible, which will correct their attachment IDs and sizes.

Finally, the legacy archives have very little information about embeds. The viewer should correctly display image embeds and a very small portion of rich embeds, but the only way to get the full embed information is by re-tracking imported messages.

<hr />

You can find the beta releases at https://github.com/chylex/Discord-History-Tracker/releases.

**Comment by TheTechRobo** [comment]
I suggest making the randomly generated server IDs a different length than the legit ones, to avoid confusion. If that would cause problems, is there a subset of IDs that don't exist at all (and never will)? I feel that'll help a bit with data integrity.

**Reply by chylex** [reply]
There are no IDs where you can guarantee they have never or will never exist. The ID 0 is at the Discord epoch, which is the beginning of 2015. If I calculated it correctly, the entire range of Discord IDs doesn't even go past this century.
